File: week3/Choi/Week3/Soy.cs
namespace Week3
{
    public class Soy : CondimentDecorator
    {
        public Beverage m_Beverage;

        public Soy(Beverage ver)
        {
            m_Beverage = ver;
        }

        public override double Cost()
        {
            return 0.15f + m_Beverage.Cost();
        }

        public override string GetDescription()
        {
            return m_Beverage.GetDescription() + " 두유";
        }
    }
}

File: week3/minj0i/Soy.java
package week3.minj0i;

public class Soy extends CondimentDecorator {
	Beverage beverage;
	
	public Soy(Beverage beverage) {
		this.beverage = beverage;
	}
	
	public String getDescription() {
		return beverage.getDescription() + ", 두유";
	}
	
	public double cost() {
		return .15 + beverage.cost();
	}
}
